neuralprophet/time_net.py

```diff
@@ -500,192 +520,174 @@ def forward_covar_net(self, covariates):
         x = x.view(x.shape[0], self.n_forecasts, len(self.quantiles))
         return x
 
-    def forward(self, inputs: Dict, meta: Dict = None, compute_components_flag: bool = False) -> torch.Tensor:
-        """This method defines the model forward pass.
-        Note
-        ----
-        Time input is required. Minimum model setup is a linear trend.
-        Parameters
-        ----------
-            inputs : dict
-                Model inputs, each of len(df) but with varying dimensions
-                Note
-                ----
-                Contains the following data:
-                Model Inputs
-                    * ``time`` (torch.Tensor , loat), normalized time, dims: (batch, n_forecasts)
-                    * ``lags`` (torch.Tensor, float), dims: (batch, n_lags)
-                    * ``seasonalities`` (torch.Tensor, float), dict of named seasonalities (keys) with their features
-                    (values), dims of each dict value (batch, n_forecasts, n_features)
-                    * ``covariates`` (torch.Tensor, float), dict of named covariates (keys) with their features
-                    (values), dims of each dict value: (batch, n_lags)
-                    * ``events`` (torch.Tensor, float), all event features, dims (batch, n_forecasts, n_features)
-                    * ``regressors``(torch.Tensor, float), all regressor features, dims (batch, n_forecasts, n_features)
-                    * ``predict_mode`` (bool), optional and only passed during prediction
-            meta : dict, default=None
-                Metadata about the all the samples of the model input batch.
-                Contains the following:
-                Model Meta:
-                    * ``df_name`` (list, str), time series ID corresponding to each sample of the input batch.
-                Note
-                ----
-                The meta is sorted in the same way the inputs are sorted.
-                Note
-                ----
-                The default None value allows the forward method to be used without providing the meta argument.
-                This was designed to avoid issues with the library `lr_finder` https://github.com/davidtvs/pytorch-lr-finder
-                while having  ``config_trend.trend_global_local="local"``.
-                The turnaround consists on passing the same meta (dummy ID) to all the samples of the batch.
-                Internally, this is equivalent to use ``config_trend.trend_global_local="global"`` to find the optimal
-                learning rate.
-            compute_components_flag : bool, default=False
-                If True, components will be computed.
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        components_stacker=ComponentStacker,
```

Comment:
components stacker should not be passed to forward, but rather inferred from a mode-flag